plugins/InspectionGadgets/InspectionGadgetsAnalysis/src/com/siyeh/ig/bugs/IncorrectDateTimeFormatInspection.java

```diff
@@ -0,0 +1,264 @@
+// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.siyeh.ig.bugs;
+
+import com.intellij.codeInspection.AbstractBaseJavaLocalInspectionTool;
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.psi.*;
+import com.siyeh.InspectionGadgetsBundle;
+import com.siyeh.ig.callMatcher.CallMatcher;
+import com.siyeh.ig.psiutils.ExpressionUtils;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import it.unimi.dsi.fastutil.ints.IntSet;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.siyeh.ig.callMatcher.CallMatcher.*;
+
+public class IncorrectDateTimeFormatInspection extends AbstractBaseJavaLocalInspectionTool {
+
+  private interface CountVerifier {
+    boolean verify(int count);
+  }
+
+  private static class RangeVerifier implements CountVerifier {
+    private final int from;
+    private final int to;
+
+    private RangeVerifier(int from, int to) {
+      this.from = from;
+      this.to = to;
+    }
+
+    @Override
+    public boolean verify(int count) {
+      return count >= from && count <= to;
+    }
+  }
+
+  private static RangeVerifier rangeOf(int from, int to) {
+    return new RangeVerifier(from, to);
+  }
+
+  private static class SetVerifier implements CountVerifier {
+    private final IntSet set;
+
+    private SetVerifier(int... numbers) {
+      set = IntOpenHashSet.of(numbers);
+    }
+
+    @Override
+    public boolean verify(int count) {
+      return set.contains(count);
+    }
+  }
+
+  private static SetVerifier setOf(int... numbers) {
+    return new SetVerifier(numbers);
+  }
+
+  private static final Map<Character, CountVerifier> DATE_TIME_FORMATTER_ALLOWED = Map.ofEntries(
+    Map.entry('G', rangeOf(1, 5)),
+    Map.entry('u', rangeOf(1, 19)),
+    Map.entry('y', rangeOf(1, 19)),
+    Map.entry('Y', rangeOf(1, Integer.MAX_VALUE)), //according to refs must be until 19, but now there is no exception
```

Comment:
You mean, according to the specification? I would say something like "according to the specification, must not exceed 19 but current implementation doesn't restrict this"

Reply:
Thank you. Replaced